modules/zip/src/parse-zip/zip-compozition.ts

```diff
@@ -111,3 +116,74 @@ export async function addOneFile(zipUrl: string, fileToAdd: ArrayBuffer, fileNam
     )
   );
 }
+
+/**
+ * creates zip archive with no compression
+ * @param inputPath path where files for the achive are stored
+ * @param outputPath path where zip archive will be placed
+ */
+export async function createZip(inputPath: string, outputPath: string) {
+  const fileIterator = await getFileIterator(inputPath);
+
+  const resFile = new NodeFile(outputPath, 'w');
+
+  const cdArray: ArrayBuffer[] = [];
+  for await (const file of fileIterator) {
+    const size = (await resFile.stat()).bigsize;
+    const [localPart, cdHeaderPart] = await generateFileHeaders(file.path, file.file, size);
+    await resFile.append(localPart);
+    cdArray.push(cdHeaderPart);
+  }
+  // TODO add hash file there
+  const cdOffset = (await resFile.stat()).bigsize;
+  const cd = concatenateArrayBuffers(...cdArray);
+  await resFile.append(new Uint8Array(cd));
+  const eoCDStart = (await resFile.stat()).bigsize;
+  await resFile.append(
+    new Uint8Array(
+      generateEoCD({recordsNumber: cdArray.length, cdSize: cd.byteLength, cdOffset, eoCDStart})
+    )
+  );
+}
+
+/**
+ * creates iterator providing buffer with file content and path to every file in the input folder
+ * @param inputPath path to the input folder
+ * @returns iterator
+ */
+export async function getFileIterator(
+  inputPath: string
+): Promise<AsyncIterable<{path: string; file: ArrayBuffer}>> {
+  const fileList = await getAllFiles(inputPath);
+  const iterable = (async function* () {
+    for (const filePath of fileList) {
+      const file = await (await fetchFile(path.join(inputPath, filePath))).arrayBuffer();
+      yield {path: filePath, file};
+    }
+  })();
+  return iterable;
+}
+
+/**
+ * creates a list of relative paths to all files in the provided folder
+ * @param basePath path of the root folder
+ * @param subfolder relative path from the root folder.
+ * @returns list of paths
+ */
+export async function getAllFiles(basePath: string, subfolder: string = ''): Promise<string[]> {
```

Comment:
```suggestion
export async function getAllFiles(fs: FileSystem, basePath: string, subfolder: string = ''): Promise<string[]> {
```
- I wonder if we could have the app pass in a `FileSystem` class here to make things a bit more portable? 
- That is, not hard code a NodeFileSystem?
-

Reply:
There is a problem, since we don't have any way in `FileSystem` to define if it's path to a directory or to a file. And this is crucial for this exact function. I think we can upgrade `FileSystem` interface and implementation later and add it there, but for now I don't see any other way to implement zip composition other than using `NodeFileSystem`